src/foam/core/stdlib.js

```diff
@@ -123,6 +123,17 @@ foam.LIB({
 });
 
 
+foam.LIB({
+  name: 'foam.Promise',
+  methods: [
+    function reduce(arry, fn, parentPromise) {
```

Comment:
A better name for parentPromise would be opt_parentPromise to indicate that it's optional.